### PR TITLE
Setup unconditioned experiment for identity, doubleclick and adsense

### PR DESCRIPTION
**Pull request by bradfrizzell**

### CODE REVIEW

File: extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js [hunk 1]
```diff
@@ -27,14 +27,20 @@ import {
...
   getExperimentBranch,
   forceExperimentBranch,
+  randomlySelectUnsetExperiments,
 } from '../../../src/experiments';
 import {dev} from '../../../src/log';
 
 /** @const {!string} @visibleForTesting */
 export const ADSENSE_A4A_EXPERIMENT_NAME = 'expAdsenseA4A';
 
+/** @const {!string} @visibleForTesting */
```
Comment (glevitzky) [inline extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js]: Nit: no need for the `!` in the type declaration.

File: extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js [hunk 2]
```diff
@@ -84,11 +96,36 @@ export function adsenseIsA4AEnabled(win, element, useRemoteHtml) {
   return true;
 }
 
+function unconditionedExperimentSelection(win, element) {
```
Comment (glevitzky) [inline extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js]: Does this need to be a function? It's only called in one place.

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 3]
```diff
@@ -113,43 +119,56 @@ export class DoubleclickA4aEligibility {
...
-        win, element,[DOUBLECLICK_UNCONDITIONED_EXPERIMENTS.FF_CANONICAL_CTL,
-          DOUBLECLICK_UNCONDITIONED_EXPERIMENTS.FF_CANONICAL_EXP],
-        DFP_UNCONDITIONED_CANONICAL_FF_EXPERIMENT_NAME);
-    if (!!experimentId) {
-      addExperimentIdToElement(experimentId, element);
-      forceExperimentBranch(
-          win, DFP_CANONICAL_FF_EXPERIMENT_NAME, experimentId);
-    }
-
+    let experimentId = this.unconditionedExperimentSelection(win, element);
```
Comment (glevitzky) [inline extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: This function doesn't return anything.

File: extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js (hunk 2 above)
Comment (glevitzky) [reply extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js]: Got it.

File: extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js [hunk 4]
```diff
@@ -65,6 +76,7 @@ export const URL_EXPERIMENT_MAPPING = {
  * @returns {boolean}
  */
 export function adsenseIsA4AEnabled(win, element, useRemoteHtml) {
+  unconditionedExperimentSelection(win, element);
   if (useRemoteHtml || !isGoogleAdsA4AValidEnvironment(win) ||
       !element.getAttribute('data-ad-client')) {
     return false;
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js]: Need to modify url based experiment selection to not attempt to select into ADSENSE_EXPERIMENT_FEATURE.IDENTITY_CONTROL/EXPERIMENT if unconditioned has been selected

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 5]
```diff
@@ -113,43 +119,69 @@ export class DoubleclickA4aEligibility {
...
-      forceExperimentBranch(
-          win, DFP_CANONICAL_FF_EXPERIMENT_NAME, experimentId);
+      forceExperimentBranch(win, expName, experimentId);
     }
+  }
 
+  /** Whether Fast Fetch is enabled
+   * @param {!Window} win
+   * @param {!Element} element
+   * @param {!boolean} useRemoteHtml
```
Comment (glevitzky) [inline extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: Hate to stickle, but we should keep the type declarations consistent: either `!boolean` or `boolean`, with the latter preferred. Same with the `!string`'s in the new doc strings you added.

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 6]
```diff
@@ -113,43 +119,69 @@ export class DoubleclickA4aEligibility {
...
     const urlExperimentId = extractUrlExperimentId(win, element);
     let experimentName = DFP_CANONICAL_FF_EXPERIMENT_NAME;
-
     if (!this.isCdnProxy(win)) {
       // Ensure that forcing FF via url is applied if test/localDev.
       if (urlExperimentId == -1 &&
           (getMode(win).localDev || getMode(win).test)) {
         experimentId = MANUAL_EXPERIMENT_ID;
       } else {
+        let e;
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: let's try to give variable names more than one character :D

File: extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js (hunk 4 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js (hunk 6 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: done

**Pull request merged**